client/views/view_units.cpp

```diff
@@ -206,9 +259,68 @@ void units_view::update_view()
     ui.units_widget->setItem(max_row, j, item_totals);
   }
 
-  ui.units_label->setText(QString(_("Units:")));
   ui.units_widget->resizeRowsToContents();
   ui.units_widget->resizeColumnsToContents();
+  update_waiting();
+}
+
+/**
+ * Function to load the units waiting table
+ */
+void units_view::update_waiting()
+{
+  int units_count = 0;
+
+  ui.uwt_widget->clearContents();
+
+  if (!client_has_player()) {
+    return;
+  }
+
+  unit_list_iterate(client_player()->units, punit)
+  {
+    if (!can_unit_move_now(punit) && punit->ssa_controller == SSA_NONE) {
+      QTableWidgetItem *item =
+          new QTableWidgetItem(utype_name_translation(punit->utype));
+      item->setData(Qt::UserRole,
+                    QVariant::fromValue(static_cast<void *>(punit)));
+      ui.uwt_widget->setItem(units_count, 0, item);
+
+      int pcity_near_dist;
+      struct city *pcity_near = get_nearest_city(punit, &pcity_near_dist);
+      ui.uwt_widget->setItem(
+          units_count, 1,
+          new QTableWidgetItem(
+              get_nearest_city_text(pcity_near, pcity_near_dist),
+              pcity_near_dist));
+
+      ui.uwt_widget->setItem(
+          units_count, 2,
+          new QTableWidgetItem(move_points_text(punit->moves_left, false),
```

Comment:
Aren't UWT units supposed to have full MP anyway?

Reply:
Let's talk about this one in #1808